Libplanet/HashDigest.cs

```diff
@@ -144,42 +145,31 @@ public static HashDigest<T> FromString(string hexDigest)
         }
 
         /// <summary>
-        /// Tests if a digest starts with the given number
-        /// (i.e., <paramref name="bits"/>) of zero bits.
-        /// <para>Note that it is bitwise, not bytes.</para>
+        /// Tests if a digest is less than the target computed for the given
+        /// <paramref name="difficulty"/>).
         /// </summary>
-        /// <param name="bits">The number of leading zero bits to test.</param>
-        /// <returns><c>true</c> only if a digest has leading zero bits of
-        /// the given number (i.e., <paramref name="bits"/>), or more than that.
-        /// If <paramref name="bits"/> is <c>0</c> it always returns
-        /// <c>true</c>.   Otherwise, it returns <c>false</c>.
+        /// <param name="difficulty">The difficulty to compute target number.
+        /// </param>
+        /// <returns><c>true</c> only if a digest is less than the target
+        /// computed for the given <paramref name="difficulty"/>).
+        /// If <paramref name="difficulty"/> is <c>0</c> it always returns
+        /// <c>true</c>.
         /// </returns>
         [Pure]
-        public bool HasLeadingZeroBits(int bits)
+        public bool LessThanTarget(long difficulty)
         {
-            int leadingBytes = bits / 8;
-            int trailingBits = (int)bits % 8;
-
-            if (ByteArray.Length < (bits / 8) + 1)
+            if (difficulty == 0)
             {
-                return false;
+                return true;
             }
 
-            for (int i = 0; i < leadingBytes; i++)
-            {
-                if (ByteArray[i] != 0)
-                {
-                    return false;
-                }
-            }
+            BigInteger target = new BigInteger(
+                Math.Pow(2, 256) / difficulty);
```

Comment:
Minor, but `Math.Pow(2, 256)` might be better treated as a constant.

Reply:
I tried to make `Math.Pow(2, 256)` to constant but it can't be determined at compile time. So I just made it as a variable.